src/Bindings/Browser/ObjectService.php

```diff
@@ -941,4 +1017,75 @@ public function updateProperties(
             }
         }
     }
+
+    /**
+     * @param string $identifier
+     * @param mixed $additionalHashValues
+     * @return array
+     */
+    protected function createCacheKey($identifier, $additionalHashValues)
+    {
+        return array(
+            $identifier,
+            sha1(is_array($additionalHashValues) ? serialize($additionalHashValues) : $additionalHashValues)
```

Comment:
I did consider adding for example Doctrine cache as dependency but am still choosing not to, mostly because our caching needs are not *that* complex. My reasoning for creating a naive array cache is that the cache only gets used from inside the ObjectService and the implementation here doesn't really suit other cases (as you mention). But, if it only gets used internally here then there should be no issues with mismatching cache keys - the additional hash values are always in the same order.

Regarding a special cache object - I don't have any major objections to that (other than it not truly being required). But I think I would prefer to do this at a later time if we end up needing a cache implementation elsewhere. For the time being, simple (and completely internal) might be better.